conans/client/graph/build_mode.py

```diff
@@ -44,9 +46,25 @@ def __init__(self, params, output):
 
             if self.never and (self.outdated or self.missing or self.patterns or self.cascade):
                 raise ConanException("--build=never not compatible with other options")
-        self._unused_patterns = list(self.patterns)
+        self._excluded_patterns = [it for it in list(self.patterns) if it.startswith("!")]
```

Comment:
Once the exclude_patterns has been computed, we don't need the ``!`` anymore. It seems it would be better to clean it directly while computing ``self._excluded_patterns``

Reply:
My idea was printing to user that a package could be wrong named. If we don't keep that list, we can't provide such information.

Reply:
What do you mean? I don't understand. We can keep that list, but in any case the ``!`` is syntactic sugar, lets call it ``--build-exclude`` and that will remove all ambiguities.

Reply:
Oh I see! Indeed if we use a new argument instead of !, we don't need to parse the value. Okay, if we will accept `--build-exclude`, I prefer accepting only it and excluding ! to avoid any ambiguous behavior.